src/lib/process.ts

```diff
@@ -0,0 +1,118 @@
+import { EventListener } from "./events";
+
+const RunService = game.GetService("RunService");
+
+export default function processHandler() {
+	const TPS = 20; // TPS to be replaced with value from tina.yaml
+	const processes = new Map<string, Process>();
+	const scheduler = new ProcessScheduler(TPS);
+
+	return (name: string) => {
+		if (processes.has(name)) {
+			return processes.get(name)!;
+		}
+		return new Process(name, scheduler);
+	};
+}
+
+/**
+ *
+ */
+export class Process extends EventListener<[]> {
+	public isSuspended = false;
+	public suspensionTime = -1;
+	public name: string;
```

Comment:
One could make a case for the order of fields;
```suggestion
	public name: string;
	public isSuspended = false;
	public suspensionTime = -1;
```